sunbird-rc-esignet-integration-impl/src/main/java/io/mosip/esignet/sunbirdrc/integration/service/SunbirdRCAuthenticationService.java

```diff
@@ -0,0 +1,183 @@
+                if(Objects.equals(authChallenge.getAuthFactorType(),"KBA")){
+                    return validateKnowledgeBasedAuth(kycAuthDto.getIndividualId(),authChallenge);
+                }
+                throw new KycAuthException("invalid_challenge_format");
+            }
+        } catch (KycAuthException e) {
+            throw e;
+        } catch (Exception e) {
+            log.error("KYC-auth failed with transactionId : {} && clientId : {}", kycAuthDto.getTransactionId(),
+                    clientId, e);
+        }
+        throw new KycAuthException(ErrorConstants.AUTH_FAILED);
+    }
+
+    @Override
+    public KycExchangeResult doKycExchange(String relyingPartyId, String clientId, KycExchangeDto kycExchangeDto)
+            throws KycExchangeException {
+        throw new KycExchangeException(ErrorConstants.NOT_IMPLEMENTED);
+    }
+
+    @Override
+    public SendOtpResult sendOtp(String relyingPartyId, String clientId, SendOtpDto sendOtpDto)
+            throws SendOtpException {
+        throw new SendOtpException(ErrorConstants.NOT_IMPLEMENTED);
+        }
+
+    @Override
+    public boolean isSupportedOtpChannel(String channel) {
+        return false;
+    }
+
+    @Override
+    public List<KycSigningCertificateData> getAllKycSigningCertificates() {
+        return new ArrayList<>();
+    }
+
+    private KycAuthResult validateKnowledgeBasedAuth(String individualId, AuthChallenge authChallenge) throws KycAuthException {
+
+        KycAuthResult  kycAuthResult= new KycAuthResult();
+        RegistrySearchRequestDto registrySearchRequestDto =new RegistrySearchRequestDto();
+        registrySearchRequestDto.setLimit(2);
+        registrySearchRequestDto.setOffset(0);
+        Map<String,Map<String,String>> filter=new HashMap<>();
+        String encodedChallenge=authChallenge.getChallenge();
+
+        byte[] decodedBytes = Base64.getUrlDecoder().decode(encodedChallenge);
+        String challenge = new String(decodedBytes, StandardCharsets.UTF_8);
+
+        try {
+            ObjectMapper objectMapper = new ObjectMapper();
+            JsonNode jsonNode = objectMapper.readTree(challenge);
+
+            String fieldDetails =env.getProperty("mosip.esignet.authenticator.sunbird-rc.auth-factor.kba.field-detailss");
+            List<FieldDetail> fieldDetailList = objectMapper.readValue(fieldDetails, new TypeReference<List<FieldDetail>>(){});
+
+            for(FieldDetail fieldDetail:fieldDetailList){
+                Map<String,String> hashMap=new HashMap<>();
+
+                if(!StringUtils.isEmpty(idField) && fieldDetail.getId().equals(idField)){
+                    hashMap.put(fieldKey,individualId);
+
+                }else{
+                    hashMap.put(fieldKey,jsonNode.get(fieldDetail.getId()).asText());
+                }
+                filter.put(fieldDetail.getId(),hashMap);
+            }
+            registrySearchRequestDto.setFilters(filter);
```

Comment:
Actually this request DTO construction passing the auth challenge can be moved to a different method to help making this method smaller. If you keep long methods, it is tough for people to follow the code. Best practice is every method should fit inside the screen, so some one can see the entire methods logic without scrolling.

Reply:
moved to new private method